feat: edit extensions documentation to include outbound/internal calls (#1878)
mia-platform/documentation: docs/console/company-configuration/extensions.md

> ## Extension Detail

Author: I wonder if it is ok to remove the sections _Extension Info_, _Extension Location_ and _Extension Visibility_, given the little amount of information included in these paragraphs, and group them in just one section.

The UI however includes a link to each of the three cards related to the aforementioned sections of the documentation. These card are in the same page.

I wonder if it is better to maintain the previous approach (multiple paragraph) in order to have a more consistent UI, or to keep going with the approach suggested in this PR.

- **Extension Info**: contains the identifying information for the extension, such as name and entry URL
- **Extension Location**: contains the information related to the positioning of the extension menu item within the Console;
- **Extension Visibility**: contains the information related to the visibility of the extension menu item within the Console;

In each card, you can also find the _Edit_ button, which allows you to modify the information contained in the card.

:::info
Just as during creation, Visibility is tied to the Destination Area set for the extension.
For example, if the Destination Area is set to _Company Overview_, the only available visibility option will be _Whole Company_.
:::

### Extension Configuration

If the current extension is of type Composer page, the badge will display the _Edit page with Composer_ button which brings the user to a page where it can edit the extension's `micro-lc` configuration.

![edit extension configuration](./img/extensions/edit-extension-configuration.png)

## Extension Communication with APIs

:::info
This section is only available for extensions of type _Composer_.
:::

> <!-- TODO: Do we have here a link with a comprehensible list of these APIs? -->

Author: ```suggestion
```

> It is also possible to configure the extension to communicate with other external APIs, as well with the Mia-Platform supported APIs.

Author: I wish to add a link to a comprehensible list of APIs here. I don't know if there's one yet, any suggestion on how better rephrase this to emphasize the concept of internal APIs would be greatly appreciated.

If you need to authenticate to this URL, keep the _Authentication required_ switch activated, then fill the _Authentication endpoint_ field with the endpoint to be called for authentication.

The _Authentication type_ field will allow to select the type of authentication to be used. Currently supported are:

- _Client credentials_ with Client ID and Client Secret
- _Basic_ with Username and Password

Once one of the available options is selected, below there will be the fields to fill in order to complete the authentication configuration, before to finally save with the _Save Changes_ button.

![Details of a configured outbound call](./img/extensions/extension-outbound-calls-detail.png)

After the creation, the outbound call configured will be shown in the page.
Details will include the name of the URL to call and the authentication type used with some useful information.
On the right side of the page there are menus to either update or delete the base URL, and change the credentials of the authentication.

In the configuration page there will also be an URL next to the _Base URL to contact your APIs_ label. This URL is the one to be used to access the APIs from your endpoint.

> <!-- TODO: Are security reasons? Should I be more specific? -->

Author: ```suggestion
```

> In fact, for security reasons, the URL to be called is not the destination URL itself, but a different one that is the one that will be used to access the APIs

Author: How can I rephrase this to better explain the security reasons behind the choice to use a Proxy instead of using the _Destination URL_ itself?

![edit extension configuration](./img/extensions/edit-extension-configuration.png)

## Extension Communication with APIs

:::info
This section is only available for extensions of type _Composer_.
:::

It is also possible to configure the extension to communicate with other external APIs, as well with the Mia-Platform supported APIs.
This could be useful for use cases such as retrieving data from an external backend, which allows it to execute other requests to the Mia-Platform APIs to have additional information to return
data to be shown in a UI created with the Composer.

This configuration is possible in the detail page of the extension that, for _Composer_ extensions only, will include two tabs over the _General_ tab:
the _Outbound calls_ tab and the _Inbound calls_ tab.

### Outbound calls

This tab allows you to configure the extension to communicate with other external APIs, by allowing to configure a base URL to be used for all the outbound calls.
It is also possible to define how to authenticate to the URL by including endpoint and the needed information for authentication.

:::info

> While it is possible to call multiple endpoints with the same base URL, it is possible to configure one outbound call for each extension.

Reviewer: ```suggestion
While it is possible to call multiple endpoints with the same base URL, you can only configure one outbound call for each extension.
```

> If you need to call multiple endpoints with different urls, you can create multiple extensions.

Reviewer: ```suggestion
If you need to call multiple endpoints with different URLs, you can create multiple extensions.
```

:::

When accessing to the _Outbound calls_ tab, if it has not been configured yet, an empty page with the button _Setup outbound calls_ will be shown.
Clicking this button will open a modal where you can configure the outbound call.

![Create a new outbound call](./img/extensions/extension-outbound-calls-create.png)

> The _Destination URL_ field is required and it is the base URL to be called.

Reviewer: ```suggestion
The _Destination URL_ field is required and it is the **Base URL** to be called.
```

If you need to authenticate to this URL, keep the _Authentication required_ switch activated, then fill the _Authentication endpoint_ field with the endpoint to be called for authentication.

The _Authentication type_ field will allow to select the type of authentication to be used. Currently supported are:

- _Client credentials_ with Client ID and Client Secret
- _Basic_ with Username and Password

> Once one of the available options is selected, below there will be the fields to fill in order to complete the authentication configuration, before to finally save with the _Save Changes_ button.

Reviewer: ```suggestion
Once one of the available options is selected, below there will be the fields to fill in order to complete the authentication configuration. When you have filled in those fields, you can proceed by saving the configuration with the _Save Changes_ button.
```

![Details of a configured outbound call](./img/extensions/extension-outbound-calls-detail.png)

After the creation, the outbound call configured will be shown in the page.

Details will include the name of the URL to call and the authentication type used with some useful information.
On the right side of the page there are menus to either update or delete the base URL, and change the credentials of the authentication.

To execute requests from the extension to the external API, you can use the URL listed in the _Base URL to contact your APIs_ label.
This automatically generated URL can be used as a proxy to the external API and, by replacing the part <YOUR_API> with the destination URL of the extension, you can

> have access to specific endpoints of the API (as example, the url `http://console.cloud.mia-platform.eu/proxy/extensions/123456789012345678901234/api/v1/users/me` will be translated in `https://www.my-external-url.com/api/v1/users/me`).

Reviewer: ```suggestion
have access to specific endpoints of the API (for instance, the URL `http://console.cloud.mia-platform.eu/proxy/extensions/123456789012345678901234/api/v1/users/me` will be translated in `https://www.my-external-url.com/api/v1/users/me`).
```

:::tip
When executing a request to the external API, the identifier of the user that executed the request will be automatically included to the request in the header `x-user-id`.

> If you are connecting to a customized service, you can use this header to identify the user that executed the request and perform further operations, as example to retrieve the user roles

Reviewer: ```suggestion
If you are connecting to a customized service, you can use this header to identify the user that executed the request and perform further operations, such as retrieving user roles
```

> using the [inbound calls](#inbound-calls) for customized operations or to save the user id in logs.

Reviewer: ```suggestion
using the [inbound calls](#inbound-calls) for customized operations or saving the user id in logs.
```

:::

### Inbound Calls

The _Inbound calls_ tab allows you to configure the extension to communicate with the Mia-Platform APIs,
using [Service Accounts](/development_suite/identity-and-access-management/manage-service-accounts.md) to leverage machine-to-machine communication.

> This feature is useful for cases where extensions that communicate with an external backend might need to perform operations on the console (such as deploy operations) or

Reviewer: ```suggestion
This feature is useful for use cases where extensions that communicate with an external backend might need to perform operations on the Mia-Platform Console (such as deploy operations) or
```
